mitodl/mit-open
# Add UserListDetails page

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open/issues/680

### Description (What does it do?)
This PR adds a new page for viewing the details of a `UserList` as well as allowing the user to re-order the items in the list the same way items can be reordered in a `LearningPath`. The code in `LearningResourceDetailsPage` was copied into a new component (`ListDetailsPage`) and abstracted. Then, `LearningResourceDetailsPage` was modified to utilize the new `ListDetailsPage` component, and the new `UserListDetailsPage` was set up to use it as well. 

### Screenshots (if appropriate):
<img width="719" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/23bfc949-8d49-4fda-a29a-3782ecebe7f4">
<img width="719" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/b6b475e4-c214-4365-999e-295c9c967ef9">
<img width="147" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/c1725dfc-6af3-47ca-a78a-a7fe7c08cb4c">
<img width="147" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/1e6581a5-b61e-4651-a810-5f278947a9d2">


### How can this be tested?
 - Spin up `mit-open` locally on this branch
 - Browse to http://localhost:8063/api/v1/userlists/ and verify that you have at least one `UserList`, which should be called "Favorites"
 - Browse to http://localhost:8063/api/v1/userlists/1/items/ and use the form at the bottom to add at least a couple `LearningResource` objects to your `UserList` using the "Raw data" form at the bottom
 - Browse to http://localhost:8063/userlists/ and click on your "Favorites" `UserList`, which should send you to http://localhost:8063/userlists/1
 - You should see a list of the `LearningResource` objects you added to the list, like in the screenshots above
 - Click the "Reorder" button, which should change the interface to the drag reordering interface like you also see in the screenshots
 - Drag around the `LearningResource` objects to change the order
 - Click "Done ordering" and verify that the objects stay in the same order
 - Refresh the page and verify that the order stays the same
 - Now we will smoke test Learning Paths, as this PR refractors the code that renders them
 - Make sure that your user is a superuser so they can edit Learning Paths
 - Add some `LearningResource` objects to a Learning Path either by going to the home page at http://localhost:8063/ and clicking the add to list buttons, or by using the API at http://localhost:8063/api/v1/learningpaths/. Create a Learning Path there and add items at http://localhost:8063/api/v1/learningpaths/x/items where x is the id of your Learning Path
 - Visit the Learning Paths page at http://localhost:8063/learningpaths/ and click the learning path you created, verifying that you see all the items that you added
 - Repat the steps from above regarding re-ordering and verify that everything works
